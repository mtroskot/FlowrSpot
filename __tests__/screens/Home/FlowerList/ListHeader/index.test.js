import React from 'react';
import { ImageBackground } from 'react-native';
import { shallow } from 'enzyme';
import FlowerListHeader from 'src/screens/Home/FlowerList/ListHeader';

describe('FlowerListHeader wrapper', () => {
  it('renders correctly, searchInput empty', () => {
    const props = {
      headerTitle: 'headerTitle',
      headerSubtitle: 'headerSubtitle',
      searchInput: '',
      placeholder: 'placeholder',
      handleInput: jest.fn(),
      clearInput: jest.fn()
    };

    const wrapper = shallow(<FlowerListHeader {...props} />);
    expect(wrapper.first().type()).toEqual(ImageBackground);
    expect(wrapper.find('ImageBackground')).toHaveLength(1);
    expect(wrapper.find('Text')).toHaveLength(2);
    expect(wrapper.find('View')).toHaveLength(1);
    expect(wrapper.find('Memo(SearchBar)')).toHaveLength(1);
    expect(
      wrapper
        .find('Text')
        .at(0)
        .props().children
    ).toEqual(props.headerTitle);
    expect(
      wrapper
        .find('Text')
        .at(1)
        .props().children
    ).toEqual(props.headerSubtitle);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('searchInput')
    ).toEqual(props.searchInput);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('handleInput')
    ).toEqual(props.handleInput);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('clearInput')
    ).toEqual(props.clearInput);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('placeholder')
    ).toEqual(props.placeholder);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly, searchInput not empty', () => {
    const props = {
      headerTitle: 'headerTitle',
      headerSubtitle: 'headerSubtitle',
      searchInput: 'searchInput',
      placeholder: 'placeholder',
      handleInput: jest.fn(),
      clearInput: jest.fn()
    };

    const wrapper = shallow(<FlowerListHeader {...props} />);
    expect(wrapper.first().type()).toEqual(ImageBackground);
    expect(wrapper.find('ImageBackground')).toHaveLength(1);
    expect(wrapper.find('Text')).toHaveLength(2);
    expect(wrapper.find('View')).toHaveLength(1);
    expect(wrapper.find('Memo(SearchBar)')).toHaveLength(1);
    expect(
      wrapper
        .find('Text')
        .at(0)
        .props().children
    ).toEqual(props.headerTitle);
    expect(
      wrapper
        .find('Text')
        .at(1)
        .props().children
    ).toEqual(props.headerSubtitle);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('searchInput')
    ).toEqual(props.searchInput);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('handleInput')
    ).toEqual(props.handleInput);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('clearInput')
    ).toEqual(props.clearInput);
    expect(
      wrapper
        .find('Memo(SearchBar)')
        .first()
        .prop('placeholder')
    ).toEqual(props.placeholder);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('interaction', () => {
  const props = {
    headerTitle: 'headerTitle',
    headerSubtitle: 'headerSubtitle',
    searchInput: '',
    placeholder: 'placeholder',
    handleInput: jest.fn(),
    clearInput: jest.fn()
  };

  const wrapper = shallow(<FlowerListHeader {...props} />);
  it('calling clearInput,handleInput, should trigger callbacks', () => {
    wrapper
      .find('Memo(SearchBar)')
      .first()
      .prop('handleInput')();
    wrapper
      .find('Memo(SearchBar)')
      .first()
      .prop('clearInput')();
    expect(props.handleInput).toHaveBeenCalledTimes(1);
    expect(props.clearInput).toHaveBeenCalledTimes(1);
  });
});
