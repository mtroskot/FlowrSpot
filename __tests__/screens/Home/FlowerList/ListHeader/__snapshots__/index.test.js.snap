// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowerListHeader wrapper renders correctly, searchInput empty 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../src/assets/images/background/home-background.png",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Light",
        "fontSize": 51.315789473684205,
        "lineHeight": 57.23684210526316,
        "marginTop": 118.42105263157895,
        "paddingHorizontal": 126.3157894736842,
        "textAlign": "center",
      }
    }
  >
    headerTitle
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Light",
        "fontSize": 31.57894736842105,
        "lineHeight": 33.55263157894737,
        "marginTop": 49.34210526315789,
        "opacity": 0.7,
        "textAlign": "center",
      }
    }
  >
    headerSubtitle
  </Text>
  <View>
    <Memo(SearchBar)
      clearInput={[MockFunction]}
      handleInput={[MockFunction]}
      placeholder="placeholder"
      searchInput=""
      viewStyle={
        Object {
          "marginBottom": 153.94736842105263,
          "marginTop": 39.473684210526315,
        }
      }
    />
  </View>
</ImageBackground>
`;

exports[`FlowerListHeader wrapper renders correctly, searchInput not empty 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../src/assets/images/background/home-background.png",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Light",
        "fontSize": 51.315789473684205,
        "lineHeight": 57.23684210526316,
        "marginTop": 118.42105263157895,
        "paddingHorizontal": 126.3157894736842,
        "textAlign": "center",
      }
    }
  >
    headerTitle
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Light",
        "fontSize": 31.57894736842105,
        "lineHeight": 33.55263157894737,
        "marginTop": 49.34210526315789,
        "opacity": 0.7,
        "textAlign": "center",
      }
    }
  >
    headerSubtitle
  </Text>
  <View>
    <Memo(SearchBar)
      clearInput={[MockFunction]}
      handleInput={[MockFunction]}
      placeholder="placeholder"
      searchInput="searchInput"
      viewStyle={
        Object {
          "marginBottom": 153.94736842105263,
          "marginTop": 39.473684210526315,
        }
      }
    />
  </View>
</ImageBackground>
`;
